From 1a1a52a51521d85b01c869f3c97abc6bea248f06 Mon Sep 17 00:00:00 2001
From: Hardik Shah <hardik.shah@ti.com>
Date: Tue, 10 Apr 2012 10:22:57 +0530
Subject: [PATCH 5/5] TI81xx capture: Changed return value to -Ve for error

IR#SDOCM00091236: Solved.

HeadLine: Return value is positive for overflow error
Description: VIDIOC_DQBUF should return negative error in case overflow is
detected on capture port. Applications generally look for negative errors and
take appropriate actions. But VIDIOC_DQBUF is returning positive error so
application is missing the error.

Signed-off-by: Hardik Shah <hardik.shah@ti.com>
---
 linux-2.6.37-psp04.04.00.01/drivers/media/video/ti81xx/ti81xxvin_main.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/linux-2.6.37-psp04.04.00.01/drivers/media/video/ti81xx/ti81xxvin_main.c b/linux-2.6.37-psp04.04.00.01/drivers/media/video/ti81xx/ti81xxvin_main.c
index f3cfdcf..9a1747e 100644
--- a/linux-2.6.37-psp04.04.00.01/drivers/media/video/ti81xx/ti81xxvin_main.c
+++ b/linux-2.6.37-psp04.04.00.01/drivers/media/video/ti81xx/ti81xxvin_main.c
@@ -114,7 +114,7 @@ static void ti81xxvin_overflow_handler(unsigned long data)
 		buf = list_first_entry(&inst->dma_queue, struct videobuf_buffer,
 				queue);
 		buf->ts = timevalue;
-		buf->state = VIDEOBUF_ERROR;
+		buf->state = -VIDEOBUF_ERROR;
 		wake_up_interruptible(&buf->done);
 		/*remove this buffer from the queue*/
 		list_del(&buf->queue);
-- 
1.7.0.4

